src/data/data.cc

```diff
@@ -663,6 +668,50 @@ uint64_t SparsePage::Push(const AdapterBatchT& batch, float missing, int nthread
   return max_columns;
 }
 
+#if defined(XGBOOST_BUILD_ARROW_SUPPORT)
+uint64_t SparsePage::Push(const data::ArrowAdapterBatch& batch, float missing, int nthread) {
+  // Set number of threads but keep old value so we can reset it after
+  const int nthreadmax = omp_get_max_threads();
+  if (nthread <= 0) nthread = nthreadmax;
+  int nthread_original = omp_get_max_threads();
+  omp_set_num_threads(nthread);
```

Comment:
Why do you need different amount of threads for all lib and this piece of code?

Reply:
Also, we can change # of threads timely by following option:
```cpp
#pragma omp parallel for schedule(static) num_threads(nthread)
...
```
It looks simpler.

Reply:
Here I'm just following the convention used by the SparsePage::Push template function, defined above this particular specialization.